.github/workflows/e2e.yml

```diff
@@ -0,0 +1,45 @@
+name: integration test
+
+on:
+  push:
+    branches:
+      - "main"
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+  pull_request:
+    branches:
+      - "main"
+
+env:
+  CARGO_TERM_COLOR: always
+  REGISTRY: ghcr.io
+
+jobs:
+  run-e2e:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: "1.18"
```

Comment:
go 1.17 has been defined in `go.mod`.

Reply:
What's its effect? I didn't see any error when running the tests

Reply:
Fine. changed to 1.17.y As containerd has increased its dependent golang version to 1.18.y.
We should consider upgrading it.  Then we won't have two kinds of build tags like:
```
//go:build !windows
// +build !windows
```